edx_django_utils/security/csp/middleware.py

```diff
@@ -0,0 +1,122 @@
+"""
+Middleware to add Content-Security-Policy and related headers.
+
+The ``content_security_policy_middleware`` middleware function can add
+``Content-Security-Policy``, ``Content-Security-Policy-Report-Only``, and
+``Reporting-Endpoints`` HTTP response headers. This functionality is
+configured by Django settings.
+"""
+import re
+
+from django.conf import settings
+from django.core.exceptions import MiddlewareNotUsed
+
+
+def _load_headers() -> dict:
+    """
+    Return a dict of headers to append to every response, based on settings.
+    """
+    # .. setting_name: CSP_ENFORCE
```

Comment:
Author's note: I wonder if we want a less generic prefix than `CSP_` for these. Some of these might conflict with Mozilla's `django-csp` library.

Reply:
@rgraber What do you think about `CSP_STATIC_` as a prefix? (`CSP_STATIC_ENFORCE` etc.) This would help avoid collisions but would also highlight that these are static values, and cannot be used for the more advanced `strict-dynamic` CSP features.